Add UDP Log Uploader, based on N1MM+ XML format

kwirk opened the pull request:
This enables sending of contactinfo and contactdelete N1MM+ XML messages, via UDP, similar to N1MM+ and other loggers.

In order to implement the delete properly, the foreign key constraint on the log_changes table has been removed, such that the contact ID can still be generated.

The contact ID is currently based on the cqrlog_main table primary key, and logbook callsign.

OH1KH commented:
HI!
Thanks for PR.
Few things before apply:

Preferences:
 - checking "enabled" does not enable other checkboxes  and address Tedit. It requires closing and reopening preferences now. Also uncheck does not disable them. Compare action to other online log settings.
  "Enable" checkbox needs "on change" event action to manipulate other other object's enables.

- I have not touched CHANGELOG. Instead I have edited src/changelog.html that generates the start popup view when uVersion.pas variables differ from current. That also opens from left frame list of help view.
Please add your line to that file to the top of change list between LI and /LI tags to be shown for users.

- I looked at n1mm hamdocs. It seem that they have tags for contest serial numbers, but how the contest messages (STXString/RTXString) are sent?
There seems to be exchangel tag pair but that does not cover sent and received strings and so must be for some other purpose.
**Without knowing N1MM** this feels weird that you can not sent contest exchanges, only serial numbers. 
How ever there are several contests that have both number and string.
But if it is so we just have to accept that.

- I can not test does this have effects to other online log uploads (as I do not use them) I have just trust that they all work as before, even when database has been altered. Have you tested them (immediately and manual uploads)?

kwirk commented:
Thanks for your feedback.

>     * checking "enabled" does not enable other checkboxes  and address Tedit. It requires closing and reopening preferences now. Also uncheck does not disable them. Compare action to other online log settings.
>       "Enable" checkbox needs "on change" event action to manipulate other other object's enables.

Fixed in d60d4e6

> 
>     * I have not touched CHANGELOG. Instead I have edited src/changelog.html that generates the start popup view when uVersion.pas variables differ from current. That also opens from left frame list of help view.
>       Please add your line to that file to the top of change list between LI and /LI tags to be shown for users.

Fixed in 0a9b58f

> 
>     * I looked at n1mm hamdocs. It seem that they have tags for contest serial numbers, but how the contest messages (STXString/RTXString) are sent?
>       There seems to be exchangel tag pair but that does not cover sent and received strings and so must be for some other purpose.
>       **Without knowing N1MM** this feels weird that you can not sent contest exchanges, only serial numbers.
>       How ever there are several contests that have both number and string.
>       But if it is so we just have to accept that.
> 

So I also struggled to find firm answer to this, but it appears from a few forums and examples that the best place for this is `exchange1` (incorrectly written as `exchangel` in online docs). I've added both the strings to this. From what I understand, the actually contents of this may vary by contest settings in N1MM+, so this is probably as close as I can get it. Changes in 60bbc8c

>     * I can not test does this have effects to other online log uploads (as I do not use them) I have just trust that they all work as before, even when database has been altered. Have you tested them (immediately and manual uploads)?

I have tested this in conjunction without CloudLog which I use, both automatically and manually updating.

kwirk commented:
Thanks @OH1KH :+1:
